### DISCUSSION TITLE
Cloud Firstore Plugin Should Support Logical Or Query

### DISCUSSION THREAD

**Discussion by maheshj01** [firebase/flutterfire#4525] [Feature request]
Currently, we can do logical And operations using the where clause  something like this 
## Logical AND
```
Firestore.instance.collection("messages")
    .where("field1", isEqualTo: "value1")
    .where("field2", isEqualTo: 'value2')
```
I would propose a similar functionality for logical OR 

## Proposal
```
Firestore.instance.collection("messages")
    .where("field1", isEqualTo: "value1")
    .orWhere("field1", isEqualTo: 'value2')
```
The current approach which works for me is using the [CombineLatestStream](https://pub.dev/documentation/rxdart/latest/rx/CombineLatestStream-class.html) from the [rxdart ](https://pub.dev/packages/rxdart)package 

## Current Approach
```
final stream1 = Firestore.instance.collection("messages")
    .where("field1", isEqualTo: "value1");
final stream2 = Firestore.instance.collection("messages")
    .where("field1", isEqualTo: "value2");

StreamBuilder(
    stream: CombineLatestStream.list([
    stream0,
    stream1,
  ]),
builder: (context, snapshot) {
  final data0 = snapshot.data[0];
  final data1 = snapshot.data[1];
})
```
I am not sure how efficient the current approach is, or what are the technical challenges involved with adding support for Logical Or query to firestore plugin I would love to hear more about it. Feel free to close this issue if this is a duplicate or have no plans for it.

**Comment by isAlmogK** [comment]
@maheshmnj is there any news on this I also need something like this

**Reply by maheshj01** [reply]
@AlmogRnD Unfortunately no updates.

**Reply by IchordeDionysos** [reply]
I bet something like this will eventually be added natively to Firestore, at which point an API design will be discussed internally at Firestore.
Until then, I don't think anything will happen in this repo to add a helper that does that. A better option would be to write a custom abstraction of Firestore where you can add this or write a helper to make it easier to run the two queries on your own.
